steve/util.py

```diff
@@ -155,17 +222,115 @@ def get(self, section, key):
+
+        :raises ConfigNotFound: if the config file can't be found
+
+        :returns: config file
+
+        """
+        self._data = yaml_load(open(self.project_file))
+        project = self._data['project']
 
-SteveConfig = IniConfig   # to allow easy switching in the future
+        # TODO: This is a little dirty since we're inserting stuff into
+        # the config file if it's not there, but so it goes.
+        try:
+            project['projectpath']
+        except KeyError:
+            project['projectpath'] = self.project_path
+        try:
+            project['jsonpath']
+        except KeyError:
+            project['jsonpath'] = os.path.join(project['projectpath'], 'json')
+        try:
+            project['yamlpath']
+        except KeyError:
+            project['yamlpath'] = os.path.join(project['projectpath'], 'yaml')
+        # If STEVE_CRED_FILE is specified in the environment or there's a
+        # cred_file in the config file, then open the file and pull the
+        # API information from there:
+        #
+        # * api_url
+        # * username
+        # * api_key
+        #
+        # This allows people to have a whole bunch of steve project
+        # directories and store their credentials in a central location.
+        cred_file = None
+        try:
+            cred_file = os.environ['STEVE_CRED_FILE']
+        except KeyError:
+            try:
+                cred_file = project['cred_file']
+            except KeyError:
+                pass
+
+        if cred_file:
+            cred_file = os.path.abspath(cred_file)
+
+            if os.path.exists(cred_file):
+                cfp = ConfigParser.ConfigParser()
+                cfp.read(cred_file)
+                project['api_url'] = cfp.get('default', 'api_url')
+                project['username'] = cfp.get('default', 'username')
+                project['api_key'] = cfp.get('default', 'api_key')
+
+        return self
+
+    def get(self, section, key):
+        try:
+            # this assumes the section is there
+            return self._data[section][key]
+        except KeyError:
+            raise NoOptionError("No option '{1}' in section '{0}'".format(
+                section, key))
+
+    _CONFIG = textwrap.dedent("""\
+    project:
+      # The name of this group of videos. For example, if this was a conference
+      # called EuroPython 2011, then you'd put:
+      # category: EuroPython 2011
+      category:
+
+      # The url for where all the videos are listed.
+      # e.g. url: http://www.youtube.com/user/PythonItalia/videos
+      url:
+
+      # The projectpath is where steve assumes subdirs if not explitly set
+      # projectpath: {projectpath}
+      # The jsonpath, if set, is where steve will look for the JSON files
+      # jsonpath: {jsonpath}
+      # The yamlpath, if set, is where steve will look for the YAML files
+      # yamlpath: {yamlpath}
+
+      # name of status file in project directory (**can** be steve.yaml)
```

Comment:
I really don't want to encourage having status information in the config file. It makes it harder for us to share configuration information.